Core/Relationships/RelationshipResolver.cs

```diff
@@ -127,23 +191,24 @@ public class RelationshipResolver
                 }
 
                 user_requested_mods.Add(module);
-                Add(module, new Relationship.UserRequested());
+                Add(module, new SelectionReason.UserRequested());
             }
 
-            // Now that we've already pre-populated modlist, we can resolve
+            // Now that we've already pre-populated the modlist, we can resolve
             // the rest of our dependencies.
 
-            foreach (CkanModule module in user_requested_mods)
+            foreach (var module in user_requested_mods)
             {
                 log.InfoFormat("Resolving relationships for {0}", module.identifier);
                 Resolve(module, options);
             }
 
             var final_modules = new List<Module>(modlist.Values);
-            final_modules.AddRange(registry.InstalledModules.Select(x => x.Module));
+            
 
             if (!options.without_enforce_consistency)
             {
+                final_modules.AddRange(installed_modules);
```

Comment:
Uh oh, I'm confused here. We're inside `AddModulesToInstall`, but we're now only adding those modules if we've been asked to enforce consistency.

This seems like the whole method would become a no-op in the case where we're not enforcing consistency, so this is either a bug, or I'm not understanding the code properly.